src/core/Akka.Remote/Transport/DotNetty/TcpTransport.cs

```diff
@@ -169,16 +171,17 @@ public TcpAssociationHandle(Address localAddress, Address remoteAddress, DotNett
 
         public override bool Write(ByteString payload)
         {
-            if (_channel.Open && _channel.IsWritable)
+            if (_channel.Open)
             {
-                var data = ToByteBuffer(payload);
-                _channel.WriteAndFlushAsync(data);
+                var data = ToByteBuffer(_channel, payload);
+                _channel.WriteAsync(data);
                 return true;
             }
             return false;
         }
 
-        private IByteBuffer ToByteBuffer(ByteString payload)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static IByteBuffer ToByteBuffer(IChannel channel, ByteString payload)
         {
             //TODO: optimize DotNetty byte buffer usage 
             // (maybe custom IByteBuffer working directly on ByteString?)
```

Comment:
Currently, this is the best possible implementation until we support System.Memory in Akka.NET Core. Tested using a bunch of others.